### ISSUE TITLE
Unhelpful error message on build failure

### ISSUE THREAD

**Issue by pharpend** [commercialhaskell/stack#341]
```
[src/snowdrift master] % stack build
Downloading lts-2.13 build plan ...
Downloaded lts-2.13 build plan.
Populating index cache, may take a moment ...
Updating package index hackage.haskell.org ...
Cloning package index ... 
Done populating index cache.
Downloading ghc-7.8.4 ...
Downloaded ghc-7.8.4.
Unpacking GHC ...
Configuring GHC ...
Installing GHC ...
Exit code ExitFailure 2 while running ["make","install"] in /tmp/stack-setup17463/ghc-7.8.4/
stack build  28.34s user 9.77s system 38% cpu 1:38.61 total
```

**Comment by snoyberg** [comment]
Fixed on master, we should cut a 0.0.3 release to help this along, we have multiple reports of this problem.

**Comment by pharpend** [comment]
I always hate leaving duplicate bug reports, sorry. I couldn't find anything of this nature on GitHub's issue search.

**Comment by snoyberg** [comment]
No worries, the previous ones were closed after people tested with master. And Github's search isn't exactly the best ;)

**Comment by pharpend** [comment]
This problem manifests itself when trying to use stack to build the git version of stack. Damn.

**Comment by DanBurton** [comment]
@pharpend In this case, you can work around the issue by having your own installation of ghc-7.8.4 on your path. (The failure isn't with building the snowdrift project, it's with installing ghc.)

**Comment by pharpend** [comment]
@DanBurton Well, that's somewhat self-defeating: my Linux distribution (Arch) ships with 7.10, and Snowdrift only supports 7.8, therefore I need stack. If I was able to install 7.8, I wouldn't use stack in the first place.

**Comment by pharpend** [comment]
Er, I'm well aware you can have two separate versions of GHC on the same computer; it's just somewhat of a nightmare to manage.

**Comment by chreekat** [comment]
Bummer. @pharpend, hopefully they'll cut a release soon and you'll be able
to bootstrap with a precompiled binary. I was really hoping that stack
would be a solution for your situation in particular!

**Comment by snoyberg** [comment]
We released 0.0.3 a few hours ago, could you try out that version and see what happens?

**Comment by pharpend** [comment]
I should have mentioned, I was able to build the git version with plain ol' `cabal`
